dataprofiler/tests/speed_tests/structured_throughput_testing.py

```diff
@@ -0,0 +1,115 @@
+################################################################################
+
+if __name__ == "__main__":
+
+    # set seed
+    random.seed(0)
+    np.random.seed(0)
+    dp.set_seed(0)
+
+
+    # load data
+    data = dp.Data('data/time_structured_profiler.csv')
+
+    # [0] allows model to be initialzied and added to labeler
+    sample_sizes = [0] + sample_sizes
+    profile_times = []
+    for sample_size in sample_sizes:
+        # setup time dict
+        
+        print(f"Evaluating sample size: {sample_size}")
+        df = data.data.sample(sample_size, replace=True).reset_index(drop=True)
+        
+        if PERCENT_TO_NAN:
+            samples_to_nan = int(len(df) * PERCENT_TO_NAN / 100)
+            for col_name in df:
+                ind_to_nan = random.sample(list(df.index), samples_to_nan)
+                df[col_name][ind_to_nan] = 'None'
+
+        # time profiling
+        start_time = time.time()
+        if  ALLOW_SUBSAMPLING:
+            profiler = dp.Profiler(df, options=options)
+        else:
+            profiler = dp.Profiler(df, samples_per_update=len(df),
+                                   options=options)
+        total_time = time.time() - start_time
+        
+        # get overall time for merging profiles
+        start_time = time.time()
+        merged_profile = profiler + profiler
+        merge_time = time.time() - start_time
+        
+        # get times for each profile in the columns
+        for profile in profiler.profile.values():
+            compiler_times = defaultdict(list)
+            
+            for compiler_name in profile.profiles:
+                compiler = profile.profiles[compiler_name]
+                inspector_times = dict()
+                for inspector_name in compiler._profiles:
+                    inspector = compiler._profiles[inspector_name]
+                    inspector_times[inspector_name] = inspector.times
+                compiler_times[compiler_name] = inspector_times
+            column_profile_time = {
+                'name': profile.name,
+                'sample_size': sample_size,
+                'total_time': total_time,
+                'column': compiler_times,
+                'merge': merge_time}
+            profile_times += [column_profile_time]
+
+        
+        print(f"COMPLETE sample size: {sample_size}")
+        print(f"Profiled in {total_time} seconds")
+        print(f"Merge in {merge_time} seconds")
+        print()
+
+    # Print dictionary with profile times
+    print("Results Saved")
+    # print(json.dumps(profile_times, indent=4))
+
+    # only works if columns all have unique names
+    times_table = pd.json_normalize(profile_times) \
+        .set_index(['name', 'sample_size']).sort_index()
+
+    # save json and times table
+    with open('structured_profiler_times.json', 'w') as fp:
+        json.dump(profile_times, fp, indent=4)
+    times_table.to_csv('structured_profiler_times.csv')
```

Comment:
I think we need to include the options (eg. subsampling or not, multiprocessing or not) saved somewhere so we can have a comprehensive record of speed tests

Reply:
Added these to the stored information.